pkg/sql/sem/builtins/builtins.go

```diff
@@ -8852,6 +8822,89 @@ func strftimeImpl() builtinDefinition {
 	)
 }
 
+func mergeStatsMetadataHelper(
+	metadata *appstatspb.AggregatedStatementMetadata, metadataDatum tree.Datum,
+) error {
+	if metadataDatum == tree.DNull {
+		return nil
+	}
+
+	var statistics appstatspb.CollectedStatementStatistics
+	metadataJSON := tree.MustBeDJSON(metadataDatum).JSON
+
+	// Only decode and set the query info if it was not previously set. Avoid the
+	// overhead of parsing the query string which can be large.
+	if metadata.Query == "" || metadata.QuerySummary == "" {
+		err := sqlstatsutil.DecodeStmtStatsMetadataJSON(metadataJSON, &statistics)
+		if err != nil {
+			return err
+		}
+		metadata.ImplicitTxn = statistics.Key.ImplicitTxn
+		metadata.Query = statistics.Key.Query
+		metadata.QuerySummary = statistics.Key.QuerySummary
+		metadata.StmtType = statistics.Stats.SQLType
+	} else {
+		err := sqlstatsutil.DecodeStmtStatsMetadataFlagsOnlyJSON(metadataJSON, &statistics)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Avoid creating the array if the db names match.
+	if len(metadata.Databases) != 1 || metadata.Databases[0] != statistics.Key.Database {
```

Comment:
Why do we default to adding the new database if there's more than 1 database? Couldn't there be a case where we have multiple databases that _include_ the new database?

Or is this solely improving the case where we only have 1 database and it's the same? (If so, is this a common case?)